src/App.js

```diff
@@ -1,85 +1,224 @@
+        });
+      } else {
+        this.setState(prev => ({
+          allTodos: [
+            ...prev.allTodos,
+            {
+              id: prev.prevId + 1,
+              title: prev.inputText.trim(),
+              status: 'active',
+              checked: false,
+            },
+          ],
+          prevId: prev.prevId + 1,
+          inputText: '',
+          todosLength: prev.todos.length + 1,
+        }));
+        this.filterCurrentValue();
+      }
+    }
+  }
+
+  changeInput = (e) => {
+    this.setState({
+      inputText: e.target.value,
+    });
+  }
+
+  filterCurrentValue = () => {
+    if (this.state.currentFilter === 'All') {
+      this.setState(prev => ({
+        todos: [...prev.allTodos],
+      }));
+    } else {
+      this.setState(prev => ({
+        todos: prev.allTodos.filter(item => (
+          item.status === prev.currentFilter.toLowerCase()
+        )),
+      }));
+    }
+  }
+
+  changeAllItemsCallback = (status, checked) => {
+    this.setState(prev => ({
+      allTodos: prev.allTodos.map((item) => {
+        const newItem = { ...item };
+
+        newItem.status = status;
+        newItem.checked = checked;
+
+        return newItem;
+      }),
+    }));
+
+    this.setState(prev => ({
+      todos: [...prev.allTodos],
+    }));
+  }
+
+  changeAllItems = (count) => {
+    if (!count) {
+      this.changeAllItemsCallback('active', false);
+    } else {
+      this.changeAllItemsCallback('completed', true);
+    }
+  }
+
+  deleteItemList = (id) => {
+    this.setState(prev => ({
+      allTodos: prev.allTodos.filter(item => item.id !== id),
+    }));
+
+    this.filterCurrentValue();
+  }
+
+  counterItemsLeft = () => (
+    this.state.allTodos.filter(item => item.status === 'active').length
+  )
+
+  switchTaskStatus = (id) => {
+    this.setState(prev => ({
+      allTodos: prev.allTodos.map((item) => {
+        const newItem = { ...item };
+
+        if (item.id === id) {
+          newItem.status = item.status === 'completed' ? 'active' : 'completed';
+          newItem.checked = !item.checked;
+        }
 
-      <footer className="footer">
-        <span className="todo-count">
-          3 items left
-        </span>
-
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
-
-          <li>
-            <a href="#/active">Active</a>
-          </li>
-
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
-
-        <button type="button" className="clear-completed">
-          Clear completed
-        </button>
-      </footer>
-    </section>
-  );
+        return newItem;
+      }),
+    }));
+
+    this.filterCurrentValue();
+  }
+
+  clearCompleted = () => {
+    this.setState(prev => ({
+      allTodos: prev.allTodos.filter(item => item.status === 'active'),
+      todos: prev.todos.filter(item => item.status === 'active'),
+    }));
+  }
+
+  filterComponents = (e, method) => {
```

Comment:
it would be better if you filter the initial array on in render method and don't store two arrays in the state